pkg/config/model/viper.go

```diff
@@ -96,6 +96,21 @@ func (s Source) IsGreaterOrEqualThan(x Source) bool {
 	return sourcesPriority[s] >= sourcesPriority[x]
 }
 
+// PreviousSource returns the source before the current one, or Unknown if there isn't one
+func (s Source) PreviousSource() Source {
+	if s == SourceDefault || s == SourceUnknown {
+		return SourceUnknown
+	} else if s == SourceFile {
+		return SourceDefault
+	}
+	for i, src := range sources {
+		if src == s {
+			return sources[i-1]
+		}
+	}
```

Comment:
We could use `sourcesPriority` maybe  ?

```
prio := sourcesPriority[s)
previousSource := sources[prio -1]
``` ?

Reply:
Yup, that works much better.